### ISSUE TITLE
Programs: something's wrong with data validation (encounter form)

### ISSUE THREAD

**Issue by richardmoizeau** [openmsupply/open-msupply#1353]
## What went wrong? 😲
 

I entered a value in the `MUAC` field and delete it (I made a mistake, was meant to enter a value for the weight instead). 

I deleted it and now it's telling me something wrong: 

![Screen Shot 2023-03-15 at 12 07 04 PM](https://user-images.githubusercontent.com/74992958/225161781-d696d111-db01-488d-b78a-0f9ddd6975a6.png)

closing and re-opening the modal does not seem to work

## Expected behaviour 🤔

- [ ] should not tell that there is an incorrect format if field is empty

## How to Reproduce 🔨

Steps to reproduce the behaviour:

Dispensary > Patients > Encounters tab > HIV Care Encounter > Physical Exam section

Enter a value and delete it. 

## Your environment 🌱
 - Version 1.1.4
 - [Platform](https://demo-open.msupply.org:88/dispensary/patients?sort=code)

**Comment by CarlosNZ** [comment]
@clemens-msupply Is there any particular reason why these are `string` values in the schema?

```
"height": {
              "description": "50373000\tBody height measure [m]",
              "examples": ["1.55", "0.92", "1.75"],
              "pattern": "^\\d\\.\\d{2}$",
              "title": "Height in m",
              "type": "string"
            },
            "midUpperArmCircumference": {
              "description": "284473002\tMid upper arm circumference (MUAC) [cm]",
              "examples": ["23.0", "21.4"],
              "pattern": "^\\d+\\.\\d+$",
              "title": "MUAC in cm",
              "type": "string"
            },
```

Seems like we should just change them to `number`?

**Comment by clemens-msupply** [comment]
> Seems like we should just change them to `number`?

Yes please, seems good to me.

**Comment by roxy-dao** [comment]
Related to #513?

**Comment by mark-prins** [comment]
I haven't run feature branch for a while.. doing so with postgres will give you this:

<img width="770" alt="Screenshot 2023-03-29 at 2 24 04 PM" src="https://user-images.githubusercontent.com/9192912/228402607-9101a359-3562-4f03-b808-9d2421df0f77.png">

Not part of this issue, still, will need looking at. I'm checking now.. just thought to raise it.

**Comment by CarlosNZ** [comment]
Fixed by #1433
